src/Features/CSharp/Portable/CodeFixes/Iterator/CSharpChangeToIEnumerableCodeFixProvider.cs

```diff
@@ -50,35 +53,41 @@ public override ImmutableArray<string> FixableDiagnosticIds
                 return null;
             }
 
-            var methodDeclarationNode = node as MethodDeclarationSyntax;
-            var typeInfo = model.GetTypeInfo(methodDeclarationNode.ReturnType, cancellationToken);
-            if (typeInfo.Type == null)
-            {
-                return null;
-            }
+            
 
-            if (typeInfo.Type.InheritsFromOrEquals(ienumerableSymbol))
+            if (type.GetBaseTypesAndThis().AsImmutable().Concat(type.AllInterfaces).Contains(t =>
+                    SymbolEquivalenceComparer.Instance.Equals(t, ienumerableSymbol)))
             {
-                if (typeInfo.Type.GetArity() != 1)
+                if (type.GetArity() != 1)
                 {
                     return null;
                 }
 
-                var typeArg = typeInfo.Type.GetTypeArguments().First();
+                var typeArg = type.GetTypeArguments().First();
                 ienumerableGenericSymbol = ienumerableGenericSymbol.Construct(typeArg);
             }
             else
             {
-                ienumerableGenericSymbol = ienumerableGenericSymbol.Construct(typeInfo.Type);
+                ienumerableGenericSymbol = ienumerableGenericSymbol.Construct(type);
             }
 
             var newReturnType = ienumerableGenericSymbol.GenerateTypeSyntax();
-            var newMethodDeclarationNode = methodDeclarationNode.WithReturnType(newReturnType);
-            root = root.ReplaceNode(methodDeclarationNode, newMethodDeclarationNode);
+            var newMethodDeclarationSyntax = (node as MethodDeclarationSyntax)?.WithReturnType(newReturnType);
+            if (newMethodDeclarationSyntax != null)
+            {
+                root = root.ReplaceNode(node, newMethodDeclarationSyntax);
+            }
+
+            var oldAccessor = (node?.Parent?.Parent as PropertyDeclarationSyntax);
+            if (oldAccessor != null)
+            {
+                root = root.ReplaceNode(oldAccessor, oldAccessor.WithType(newReturnType));
+            }
+
             var newDocument = document.WithSyntaxRoot(root);
```

Comment:
If neither the method declaration case nor the property declaration case are hit, should we still return a CodeAction that does nothing?